agoradatatools/etl/transform.py

```diff
@@ -137,6 +151,105 @@ def transform_network(datasets: dict):
 
     return merged
 
+
+def transform_gene_metadata(datasets: dict, adjusted_p_value_threshold, protein_level_threshold):
+    '''
+    This function will perform transformations and incrementally create a dataset called gene_metadata.
+    Each dataset will be left_joined onto gene_info.
+    '''
+    gene_info = datasets['syn25953363']
+    igap = datasets['syn12514826']
+    eqtl = datasets['syn12514912']
+    proteomics = datasets['syn18689335']
+    brain_expression_change = datasets['syn11914808']
+    rna_change = datasets['syn14237651']
+
+    # remove duplicate ensembl_gene_ids and select columns
+    gene_info = gene_info.groupby('ensembl_gene_id').apply(lambda x: x.nlargest(1, "_version")).reset_index(drop=True)
+    gene_info = gene_info[['ensembl_gene_id', 'symbol', 'name', 'summary', 'alias']]
+
+
+    gene_metadata = pd.merge(left=gene_info, right=igap, how='left', on='ensembl_gene_id')
+    gene_metadata['igap'] = gene_metadata.apply(lambda row: False if row['hgnc_symbol'] is NaN else True, axis=1)
+    gene_metadata['igap'].fillna(False, inplace=True)
+    gene_metadata = gene_metadata[['ensembl_gene_id', 'symbol', 'name', 'summary', 'alias', 'igap']]
+
+    gene_metadata = pd.merge(left=gene_metadata, right=eqtl, how='left', on='ensembl_gene_id')
+    gene_metadata = gene_metadata[['ensembl_gene_id', 'symbol', 'name', 'summary', 'alias', 'igap', 'haseqtl']]
+    gene_metadata.rename(columns={'haseqtl': 'eqtl'}, inplace=True)
+    gene_metadata['eqtl'] = gene_metadata['eqtl'].replace({'TRUE': True}).fillna(False)
+
+    rna_change = rna_change[['ensembl_gene_id', 'adj_p_val']]
+    rna_change = rna_change.groupby('ensembl_gene_id')['adj_p_val'].agg('min').reset_index()
+
+    gene_metadata = pd.merge(left=gene_metadata, right=rna_change, how='left', on='ensembl_gene_id')
+    gene_metadata['adj_p_val'] = gene_metadata['adj_p_val'].fillna(-1)
+    gene_metadata['rna_brain_change_studied'] = gene_metadata.apply(
+        lambda row: False if row['adj_p_val'] == -1 else True, axis=1)
```

Comment:
Could this just be

```
        lambda row: row['adj_p_val'] != -1, axis=1)
```

Reply:
Yeah, I thought the explicit verbose here was easier to understand.  But it could be!